core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala

```diff
@@ -298,19 +319,79 @@ object RangeEventDescriptorV0TLV extends TLVFactory[RangeEventDescriptorV0TLV] {
   override def fromTLVValue(value: ByteVector): RangeEventDescriptorV0TLV = {
     val iter = ValueIterator(value)
 
+    val nonce = SchnorrNonce(iter.take(32))
     val start = Int32(iter.takeBits(32))
-    val stop = Int32(iter.takeBits(32))
+    val count = UInt32(iter.takeBits(32))
     val step = UInt16(iter.takeBits(16))
 
-    RangeEventDescriptorV0TLV(start, stop, step)
+    RangeEventDescriptorV0TLV(nonce, start, count, step)
+  }
+}
+
+case class LargeRangeEventDescriptorV0TLV(
+    base: UInt16,
+    isSigned: Boolean,
```

Comment:
nit: Maybe represent this as an ADT rather than a simple `Boolean` flag? The reason I think this might be useful is: 

What kind of signed number representation are you using? Twos complement? Ones complement? Some other exotic scheme?

https://en.wikipedia.org/wiki/Signed_number_representations

Reply:
I guess there needs to at least be documentation on what sort of signed number representation this actually represents?

Reply:
I think would be more of a question for the spec. In the spec it is a boolean so we should do the same here

Reply:
Ok, you are more familar with the spec work than I am. Can you file an issue and link to this comment? Since you aren't documenting where you are getting this stuff from it's very hard to cross reference and file issues like this above. Where is the signed stuff?

Reply:
https://github.com/discreetlogcontracts/dlcspecs/pull/55/files#diff-c15525daf7798c0dd4de183850e348ffdd9aea9812ebe5bc95a1c4189122a14dR90

I don't think the different representations apply here. You don't sign the whole number, you sign the base 10 version of each digit